src/controllers/GetPosts.ts

```diff
@@ -0,0 +1,52 @@
+import { Response, Request } from 'express'
+import { User, Post, Like, Tag, Animal } from '../db'
+import { Op } from 'sequelize'
+
+const getPosts = async (req: Request, res: Response) => {
+  const { tagId, animalId, searchCountent } = req.query
```

Comment:
it's a common convention to use `q` in query string for searching

```suggestion
  const { tagId, animalId, q } = req.query
```